app/tests/spec/lib/validate.js

```diff
@@ -134,5 +134,112 @@ define(function (require, exports, module) {
         assert.isTrue(Validate.isPasswordValid(createRandomHexString(Constants.PASSWORD_MIN_LENGTH)));
       });
     });
+
+    describe('isDataValid', function () {
+      /*eslint-disable sorting/sort-object-props */
+
+      it('is defined', function () {
+        assert.isFunction(Validate.isDataValid);
+      });
+
+      it('expects two arguments', function () {
+        assert.lengthOf(Validate.isDataValid, 2);
+      });
+
+      it('returns true when simple type matches the schema', function () {
+        assert.isTrue(Validate.isDataValid('foo', 'String'));
+      });
+
+      it('returns false when simple type does not match the schema', function () {
+        assert.isFalse(Validate.isDataValid('foo', 'Number'));
+      });
+
+      it('handles null', function () {
+        assert.isTrue(Validate.isDataValid(null, 'Null'));
+      });
+
+      it('handles undefined', function () {
+        assert.isTrue(Validate.isDataValid(undefined, 'Undefined'));
+      });
+
+      it('handles false', function () {
+        assert.isTrue(Validate.isDataValid(false, 'Boolean'));
+      });
+
+      it('handles zero', function () {
+        assert.isTrue(Validate.isDataValid(0, 'Number'));
+      });
+
+      it('handles empty string', function () {
+        assert.isTrue(Validate.isDataValid('', 'String'));
+      });
+
+      it('returns true when complex type matches the schema', function () {
+        assert.isTrue(Validate.isDataValid(
+          [ 'foo', 3.14159265359, {} ],
+          [ 'String', 'Number', 'Object' ]
+        ));
+      });
+
+      it('returns false when complex type does not match the schema', function () {
+        assert.isFalse(Validate.isDataValid(
+          [ 'foo', 3.14159265359, [] ],
+          [ 'String', 'Number', 'Object' ]
+        ));
+      });
+
+      it('handles different key order', function () {
+        assert.isTrue(Validate.isDataValid(
+          { foo: 'bar', baz: 'qux' },
+          { baz: 'String', foo: 'String' }
+        ));
+      });
+
+      it('handles nested objects', function () {
+        assert.isTrue(Validate.isDataValid(
+          { foo: { bar: [ 'baz', 'qux' ] } },
+          { foo: { bar: [ 'String', 'String' ] } }
+        ));
+        assert.isFalse(Validate.isDataValid(
+          { foo: { bar: [ 'baz', 'qux' ] } },
+          { foo: { bar: [ 'String', 'Number' ] } }
+        ));
+      });
+
+      it('ignores undefined properties if schema property is undefined', function () {
+        assert.isTrue(Validate.isDataValid(
+          { foo: 'bar', baz: undefined },
```

Comment:
> Fx developers that receive payloads with undefined values have good reason to question, "is everything legit here?"

So maybe the resolution is to strip those before validation and leave the schema as is? I'm torn.